code/frameworks/react-vite/src/plugins/react-docgen.ts

```diff
@@ -36,14 +36,23 @@ export function reactDocgen({
     name: 'storybook:react-docgen-plugin',
     enforce: 'pre',
     async transform(src: string, id: string) {
-      const relPath = path.relative(cwd, id);
-      if (!filter(relPath)) return;
+      if (!filter(path.relative(cwd, id))) {
+        return;
+      }
 
       try {
         const docgenResults = parse(src, {
           resolver: defaultResolver,
           handlers,
-          importer: defaultImporter,
+          importer: makeFsImporter((filename, basedir) => {
+            const result = defaultLookupModule(filename, basedir);
+
+            if (!result.match(/\.(mjs|tsx?|jsx?)$/)) {
```

Comment:
Why don't we use the `RESOLVE_EXTENSIONS`?

Reply:
Because I though this would be more expressive.

Reply:
We can write it out, but we should make sure that all possible extensions which the user might use are supported (optional <c|m> prefix for all <js|ts> files with optional <x> suffix).

Reply:
Sure!

Reply:
OK.